src/python/pants/option/global_options.py

```diff
@@ -125,22 +143,26 @@ def is_available(self) -> bool:
 class DynamicRemoteOptions:
     """Options related to remote execution of processes which are computed dynamically."""
 
-    remote_execution: bool
-    remote_cache_read: bool
-    remote_cache_write: bool
-    remote_instance_name: str | None
-    remote_store_headers: dict[str, str]
-    remote_execution_headers: dict[str, str]
+    execution: bool
+    cache_read: bool
+    cache_write: bool
+    instance_name: str | None
+    store_address: str | None
+    execution_address: str | None
+    store_headers: dict[str, str]
+    execution_headers: dict[str, str]
```

Comment:
nit: not sure a dict can actually be part of a frozen data class.. since a dict is mutable...
not super important, but worth paying attention to.

Reply:
Yeah `hash()` will fail. I'm not concerned enough to go through the ceremony of a custom `__init__()` and `@frozen_after_init` etc, given that this does not use the Rules API.